sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -152,24 +153,34 @@ class OrcFileFormat
       assert(supportBatch(sparkSession, resultSchema))
     }
 
-    OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE.setBoolean(hadoopConf, sqlConf.caseSensitiveAnalysis)
-
-    val broadcastedConf =
-      sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
     val isCaseSensitive = sparkSession.sessionState.conf.caseSensitiveAnalysis
+    OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE.setBoolean(hadoopConf, isCaseSensitive)
+    val broadcastedHadoopConf = if (options.isEmpty) {
+      Option.empty
+    } else {
+      Option(sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf)))
+    }
+
     val orcFilterPushDown = sparkSession.sessionState.conf.orcFilterPushDown
+    val sparkConf = sparkSession.sparkContext.conf
 
     (file: PartitionedFile) => {
-      val conf = broadcastedConf.value.value
+      val sharedConf = if (broadcastedHadoopConf.isDefined) {
+        broadcastedHadoopConf.get.value.value
+      } else {
+        val conf = SparkHadoopUtil.newConfiguration(sparkConf)
```

Comment:
You are assuming `hadoopConf` == `SparkHadoopUtil.newConfiguration(sparkConf)`
This does not hold in general, particularly when the code evolves.